compiler/semtypes.nim

```diff
@@ -1527,7 +1539,10 @@ proc semGeneric(c: PContext, n: PNode, s: PSym, prev: PType): PType =
     return errorType(c)
   if tx != result and tx.kind == tyObject:
     if tx[0] != nil:
-      semObjectTypeForInheritedGenericInst(c, n, tx)
+      if not trySemObjectTypeForInheritedGenericInst(c, n, tx):
+        {.cast(uncheckedAssign).}:
+          result.kind = tyError
```

Comment:
Dangerous, why is it required? Why can't it be `return newOrPrevType(tyError, prev, c)` or similar?

Reply:
It's not, I'm just daft.